### PR TITLE
chore(bitbucket): [FS-1008]: add an allowlist for bitbucket

### PR DESCRIPTION
**Pull request by vivekkhimani**
As we are officially adding BBDC support, we are expecting customers to configure it using network broker. Getting ahead and adding a dedicated allowlist for bitbucket.

### CODE REVIEW

File: pkg/config.go [hunk 1]
```diff
@@ -224,6 +229,7 @@ type InboundProxyConfig struct {
 	Heartbeat       HeartbeatConfig  `mapstructure:"heartbeat" json:"heartbeat"`
 	GitHub          *GitHub          `mapstructure:"github" json:"github"`
 	GitLab          *GitLab          `mapstructure:"gitlab" json:"gitlab"`
+	BitBucket          *BitBucket          `mapstructure:"bitbucket" json:"bitbucket"`
```
Comment (tpetr) [inline pkg/config.go]: spacing looks weird here, can you run gofmt?

File: pkg/config.go [hunk 2]
```diff
@@ -435,5 +441,59 @@ func LoadConfig(configFiles []string, deploymentId int) (*Config, error) {
...
+
+		bitBucketBaseUrl, err := url.Parse(bitBucket.BaseURL)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse bitbucket base URL: %v", err)
+		}
+
+		headers := map[string]string{
+			"PRIVATE-TOKEN": bitBucket.Token,
+		}
```
Comment (tpetr) [inline pkg/config.go]: can you update this so that we only set this header if `bitBucket.Token` is not an empty value? (e.g. `""`)

**Review by tpetr — DISMISSED** [review]
2 nits but lgtm overall

**Comment by vivekkhimani** [comment]
image

gas lighting

**Review by tpetr — DISMISSED** [review]

File: pkg/config.go [hunk 3]
```diff
@@ -224,6 +229,7 @@ type InboundProxyConfig struct {
 	Heartbeat       HeartbeatConfig  `mapstructure:"heartbeat" json:"heartbeat"`
 	GitHub          *GitHub          `mapstructure:"github" json:"github"`
 	GitLab          *GitLab          `mapstructure:"gitlab" json:"gitlab"`
+	BitBucket       *BitBucket          `mapstructure:"bitbucket" json:"bitbucket"`
```
Comment (tpetr) [inline pkg/config.go]: spacing here still seems weird -- those backticks should all be on the same line. did gofmt not catch this?

Comment (vivekkhimani) [reply pkg/config.go]: yeah gofmt missed it.... let me retry it

**Review by tpetr — APPROVED** [review]

**Pull request merged**